zulip/integrations/git/post-receive

```diff
@@ -37,15 +38,21 @@ def git_repository_name() -> Text:
         return os.path.basename(os.path.dirname(os.getcwd()))
 
 def git_commit_range(oldrev: str, newrev: str) -> str:
+    remote_repo_cmd = ["git", "config", "--get", "remote.origin.url"]
+    remote_repo_url = subprocess.check_output(remote_repo_cmd).strip().decode("utf-8")
+
     log_cmd = ["git", "log", "--reverse",
-               "--pretty=%aE %H %s", "%s..%s" % (oldrev, newrev)]
+               "--pretty=-[AUTHOR]=%aN-[COMMIT]=%H-[SUBJECT]=%s", "%s..%s" % (oldrev, newrev)]
     commits = ''
     for ln in subprocess.check_output(log_cmd, universal_newlines=True).splitlines():
-        author_email, commit_id, subject  = ln.split(None, 2)
-        if hasattr(config, "format_commit_message"):
-            commits += config.format_commit_message(author_email, subject, commit_id)
-        else:
-            commits += '!avatar(%s) %s\n' % (author_email, subject)
+        author, commit_sha, subject  = git.extract_commit_data(ln)
+        commits += git.COMMIT_ROW_TEMPLATE.format(
+            user_name=author,
+            commit_short_sha=commit_sha[:7],
```

Comment:
Let Git compute the abbreviation length with `%h` instead of using a fixed length. In large repositories, it automatically increases the abbreviation length to avoid collisions.

Reply:
@andersk that makes sense, thank you.
I think we need to find a different way for `get_short_sha` function  in [this file](https://github.com/zulip/zulip/blob/master/zerver/lib/webhooks/git.py) too in zulip/zulip git webhook.

Reply:
The webhook has less information since it doesn’t have the repository available to it locally.